mono/utils/mono-dl-windows.c

```diff
@@ -68,7 +72,11 @@ mono_dl_open_file (const char *file, int flags)
 		if (!hModule)
 			SetLastError (last_error);
 	} else {
+#if G_HAVE_API_SUPPORT(HAVE_CLASSIC_WINAPI_SUPPORT)
 		hModule = GetModuleHandle (NULL);
+#else
```

Comment:
Maybe GetModuleHandleEx?

Reply:
Is there an advantage to `GetModuleHandleEx`?

Reply:
When the parameter is NULL, no. When the parameter is other than NULL, GetModuleHandle implies a race condition. It is documented. It doesn't increment the reference count on the module. GetModuleHandleEx adds a reference count and closes the race condition -- up to you to FreeLibrary it. The NULL parameter means the .exe however and FreeLibrary against the .exe does nothing -- you cannot unload it, except via ExitProcess/TerminateProcess. I added GetModuleHandleEx in XP for this reason. Ex also has some hand other flags like taking an address within the module to gets its modules, and a flag to emulate the old racy API, and a flag to "pin" a module so it won't unload -- you do this in your DllMain if you have some impossible to properly cleanup resource.